File: Web.Public/src/Models/Address.cs
using System.Collections.Generic;
using System.Text.Json.Serialization;
using Web.Public.JsonObjects;

namespace Web.Public.Models
{
	public class Address : IJsonObject
	{
		[JsonIgnore]
		public long ID { get; set; }

		[JsonPropertyName("url")]
		public string URL => $"/acvp/v1/vendors/{OrganizationID}/addresses/{ID}";

		[JsonIgnore]
		public long OrganizationID { get; set; }

		[JsonPropertyName("street1")]
		public string Street1 { get; set; }

		[JsonPropertyName("street2")]
		public string Street2 { get; set; }

		[JsonPropertyName("street3")]
		public string Street3 { get; set; }

		[JsonPropertyName("locality")]
		public string Locality { get; set; }

		[JsonPropertyName("region")]
		public string Region { get; set; }

		[JsonPropertyName("country")]
		public string Country { get; set; }

		[JsonPropertyName("postalCode")]
		public string PostalCode { get; set; }

		public List<string> ValidateObject()
		{
			throw new System.NotImplementedException();
		}
	}
}

File: Web.Admin/ClientApp/src/app/models/address/Address.ts
export class Address {

  id: number;
  street1: string;
  street2: string;
  street3: string;
  locality: string;
  region: string;
  country: string;
  postalCode: string;

  public constructor(id: number,
    street1: string,
    street2: string,
    street3: string,
    locality: string,
    region: string,
    country: string,
    postalCode: string) {
    this.id = id;
    this.street1 = street1;
    this.street2 = street2;
    this.street3 = street3;
    this.locality = locality;
    this.region = region;
    this.country = country;
    this.postalCode = postalCode;
  };

}
